### ISSUE TITLE
Title: WCAG 4.1.2: Ensures every form element has a label (.PrivateSwitchBase-input-317[value=""][data-indeterminate="false"])

### ISSUE THREAD

**Issue by mihirprakash** [material-table-core/core#846]
Issue: Ensures every form element has a label (label - https://accessibilityinsights.io/info-examples/web/label)

Target application: Luminate Portal - http://localhost:7777/local-dev/plans/663b1af80f7ed815d83846bf/shipments

Element path: #mfe;.PrivateSwitchBase-input-317[value=""][data-indeterminate="false"]

Snippet: <input class="PrivateSwitchBase-input-317" type="checkbox" data-indeterminate="false" value="">

How to fix: 
Fix any of the following:
  Form element does not have an implicit (wrapped) <label>
  Form element does not have an explicit <label>
  aria-label attribute does not exist or is empty
  aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
  Element has no title attribute
  Element has no placeholder attribute
  Element's default semantics were not overridden with role="none" or role="presentation"

Environment: Chrome version 124.0.0.0

====

This accessibility issue was found using Accessibility Insights for Web 2.41.0 (axe-core 4.7.2), a tool that helps find and fix accessibility issues. Get more information & download this tool at http://aka.ms/AccessibilityInsights.
<img width="960" alt="mat-table" src="https://github.com/material-table-core/core/assets/16774320/717dfcb5-e033-457d-9b38-d55083d18a19">

**Comment by mihirprakash** [comment]
Version: "@material-table/core": "^6.2.1"

**Comment by Domino987** [comment]
Ok a few things.
Local host is your computer. That cannot be access by anybody else.

Can you verify it with the newest version.

Can you add the code/reproducible example in a code sandbox with just the code that does not work?
